gui/vm/forms.py

```diff
@@ -52,13 +59,32 @@ def clean_vcpus(self):
             else:
                 return vcpus
 
+    def clean_memory(self):
+        memory = self.cleaned_data.get('memory')
+        vm_type = self.cleaned_data.get('vm_type')
+        if vm_type == 'Container Provider' and memory < 1024:
+            raise forms.ValidationError(_('Minimum memory for container must be 1024.'))
```

Comment:
Needs units and can be rearranged a bit for clarity:
'Minimum container memory is 1,024 MiB.'